xrpl/utils/txn_parser/utils/transaction.py

```diff
@@ -0,0 +1,183 @@
+    }
+    normalized_txn["meta"] = transaction_data["meta"]
+    normalized_txn["ledger_index"] = transaction_data["ledger_index"]
+    return cast(RawTxnType, normalized_txn)
+
+
+def normalize_fields(
+    fields: Dict[str, Any],
+) -> NormalizedFields:
+    """
+    Normalize 'NewFields', 'FinalFields' and 'ModifiedFields'.
+
+    Args:
+        fields: A dictionary of all fields that field state has.
+            E.g.: {'Balance': '62537659', 'OwnerCount': 16, 'Sequence': 67702065}
+
+    Returns:
+        The full field state with all its fields as objects.
+    """
+    balance = fields["Balance"] if "Balance" in fields else None
+    low_limit = fields["LowLimit"] if "LowLimit" in fields else None
+    high_limit = fields["HighLimit"] if "HighLimit" in fields else None
+    taker_gets = fields["TakerGets"] if "TakerGets" in fields else None
+    taker_pays = fields["TakerPays"] if "TakerPays" in fields else None
+    account = cast(str, fields["Account"]) if "Account" in fields else None
+    sequence = cast(int, fields["Sequence"]) if "Sequence" in fields else None
+    flags = cast(int, fields["Flags"]) if "Flags" in fields else None
+    expiration = cast(int, fields["Expiration"]) if "Expiration" in fields else None
+    field_state = NormalizedFields(
+        Balance=AccountBalance(
+            issuer=balance["issuer"],
+            currency=balance["currency"],
+            value=balance["value"],
+        )
+        if isinstance(balance, dict)
+        else balance,
+        LowLimit=AccountBalance(
+            issuer=low_limit["issuer"],
+            currency=low_limit["currency"],
+            value=low_limit["value"],
+        )
+        if isinstance(low_limit, dict)
+        else low_limit,
+        HighLimit=AccountBalance(
+            issuer=high_limit["issuer"],
+            currency=high_limit["currency"],
+            value=high_limit["value"],
+        )
+        if isinstance(high_limit, dict)
+        else high_limit,
+        TakerGets=AccountBalance(
+            issuer=taker_gets["issuer"],
+            currency=taker_gets["currency"],
+            value=taker_gets["value"],
+        )
+        if isinstance(taker_gets, dict)
+        else taker_gets,
+        TakerPays=AccountBalance(
+            issuer=taker_pays["issuer"],
+            currency=taker_pays["currency"],
+            value=taker_pays["value"],
+        )
+        if isinstance(taker_pays, dict)
+        else taker_pays,
+        Account=account,
+        Sequence=sequence,
+        Flags=flags,
+        Expiration=expiration,
+    )
+    return field_state
+
+
+def normalize_node(affected_node: Dict[str, Any]) -> NormalizedNode:
+    """
+    Affected node to a standard format.
+
+    Args:
+        affected_node: Affected node.
+
+    Returns:
+        NormalizedNode object.
+    """
+    diff_type = cast(
+        Literal["ModifiedNode", "CreatedNode", "DeletedNode"],
+        list(affected_node.keys())[0],
```

Comment:
What are you trying to get here? Why only the first one?

Reply:
This is getting the type of the affected node (ModifiedNode, CreatedNode, DeletedNode). The types are relevant for the `parse_order_book_changes` parser to determine whether an Offer was created, canceled, filled or partially-filled.
I made it an argument so it gets explained in the docstring :) To put in the arguments please check out lines 157 - 171 of my next commit. Maybe you know a more elegant solution.